### ISSUE TITLE
Batch Normalization layers are not correctly broadcasted to nodes

### ISSUE THREAD

**Issue by sharkovsky** [fedbiomed/fedbiomed#903]
Batch Normalization layers are not correctly passed to the Nodes when loading a breakpoint or loading a pretrained model. 

Consider the following Minimal Working Example (MWE) that simulates loading a pretrained model (it works with MNIST dataset):
```python
import torch
import torch.nn as nn
from fedbiomed.common.training_plans import TorchTrainingPlan
from fedbiomed.common.data import DataManager
from torchvision import datasets, transforms


# Here we define the model to be used.
# You can use any class name (here 'Net')
class MyTrainingPlan(TorchTrainingPlan):

    # Defines and return model
    def init_model(self, model_args):
        return self.Net(model_args = model_args)

    # Defines and return optimizer
    def init_optimizer(self, optimizer_args):
        return torch.optim.Adam(self.model().parameters(), lr = optimizer_args["lr"])

    # Declares and return dependencies
    def init_dependencies(self):
        deps = ["from torchvision import datasets, transforms"]
        return deps

    class Net(nn.Module):
        def __init__(self, model_args):
            super().__init__()
            self.conv1 = nn.Conv2d(1, 32, 3, 1)
            self.conv2 = nn.Conv2d(32, 64, 3, 1)
            self.conv2_bn = nn.BatchNorm2d(64)
            self.fc1 = nn.Linear(9216, 128)
            self.fc1_bn = nn.BatchNorm1d(128)
            self.fc2 = nn.Linear(128, 10)

        def forward(self, x):
            x = self.conv1(x)
            x = F.relu(x)
            x = self.conv2(x)
            x = self.conv2_bn(x)
            x = F.relu(x)
            x = F.max_pool2d(x, 2)
            x = torch.flatten(x, 1)
            x = self.fc1(x)
            x = self.fc1_bn(x)
            x = F.relu(x)
            x = self.fc2(x)

            output = F.log_softmax(x, dim=1)
            return output

    def training_data(self):
        # Custom torch Dataloader for MNIST data
        transform = transforms.Compose([transforms.ToTensor(),
        transforms.Normalize((0.1307,), (0.3081,))])
        dataset1 = datasets.MNIST(self.dataset_path, train=True, download=False, transform=transform)
        train_kwargs = { 'shuffle': True}
        return DataManager(dataset=dataset1, **train_kwargs)

    def training_step(self, data, target):
        from fedbiomed.common.logger import logger
        logger.info(f"NODE SIDE CONV LAYER {self.model().state_dict()['conv2_bn.weight'][:5]}")
        logger.info(f"NODE SIDE BN LAYER {self.model().state_dict()['conv2_bn.running_mean'][:5]}")
        output = self.model().forward(data)
        loss   = torch.nn.functional.nll_loss(output, target)
        return loss

model_args = {}

training_args = {
    'loader_args': { 'batch_size': 1, },
    'optimizer_args': {
        "lr" : 1e-3
    },
    'dry_run': False,
    'num_updates': 1,
    'share_persistent_buffers': True  # this is True by default, but just to be explicit
}

from fedbiomed.researcher.experiment import Experiment
from fedbiomed.researcher.aggregators.fedavg import FedAverage

tags =  ['#MNIST', '#dataset']
rounds = 1

exp = Experiment(tags=tags,
                 model_args=model_args,
                 training_plan_class=MyTrainingPlan,
                 training_args=training_args,
                 round_limit=rounds,
                 aggregator=FedAverage(),
                 node_selection_strategy=None)


## SIMULATE LOADING A PRETRAINED MODEL
state_dict = exp.job().training_plan.model().state_dict()
state_dict['conv2_bn.running_mean'][0] += 99.
state_dict['conv2_bn.weight'][0] += 99.
exp.training_plan().set_model_params(state_dict)
exp.job().update_parameters()

print('********************************************************')
print(f"RESEARCHER SIDE CONV LAYER {state_dict['conv2_bn.weight'][:5]}")
print(f"RESEARCHER SIDE BN LAYER {state_dict['conv2_bn.running_mean'][:5]}")
print('********************************************************')

exp.run()
```

The expected output is that the researcher side values and node side values should be the same. That is indeed true for the convolutional layer, but is incorrect for the batch norm layer. 

This goes against the researcher's explicit request of `'share_persistent_buffers': True`.

**Comment by sharkovsky** [comment]
The issue is in the [`job.update_parameters`](https://github.com/fedbiomed/fedbiomed/blob/2e4fc2fdfc7843a966aebe195e8d61fbab178188/fedbiomed/researcher/job.py#L718) function, where we use `get_model_params` which only returns the trainable weights. 

One solution would be to use `after_training_params` which correctly takes into account the `share_persistent_buffers` option, but is not ideal because the intended use of this function was for correcting the parameters after the local training. 
A second solution would be to take the `share_persistent_buffers` option into account also in the `get_model_params` function, but I think the original intended use of this function was specifically to return only the trainable parameters.

@pandrey-fr do you remember our discussions about this? Have any comment?

**Comment by pandrey-fr** [comment]
I have to look back at this; I remember that we had a discussion on whether/how BatchNorm parameters should be shared during training (and that there turned out to be some literature on the topic). As to the question of checkpointing them, at the moment I have no specific memory...

Note to myself (@pandrey-fr) to go back to this next week.
